include/messages.h

```diff
@@ -341,16 +364,23 @@ struct MLSPlaintext
 // } MLSCiphertext;
 struct MLSCiphertext
 {
-  tls::opaque<1> group_id;
+  bytes group_id;
   epoch_t epoch;
   ContentType content_type;
-  tls::opaque<1> sender_data_nonce;
-  tls::opaque<1> encrypted_sender_data;
-  tls::opaque<4> authenticated_data;
-  tls::opaque<4> ciphertext;
+  bytes sender_data_nonce;
+  bytes encrypted_sender_data;
+  bytes authenticated_data;
+  bytes ciphertext;
 
   TLS_SERIALIZABLE(group_id, epoch, content_type, sender_data_nonce,
                    encrypted_sender_data, authenticated_data, ciphertext);
+  TLS_TRAITS(tls::vector<1>,
+             tls::pass,
+             tls::pass,
+             tls::vector<1>,
```

Comment:
don't we need tls::variant<ContentType> here (similar to line 353)

Reply:
No, because the content_type here is separated from the actual content. tls::variant is really intended for the case where you have the selector value followed immediately by the selected type.